apps/web/src/hooks/useThreadsStatus.ts

```diff
@@ -0,0 +1,192 @@
+import useSWR from "swr";
+import { ThreadUIStatus, ThreadStatusData } from "@/lib/schemas/thread-status";
+import { fetchThreadStatus } from "@/services/thread-status.service";
+import { THREAD_STATUS_SWR_CONFIG } from "@/lib/swr-config";
+import { useMemo, useRef } from "react";
+import { Thread } from "@langchain/langgraph-sdk";
+import { ManagerGraphState } from "@open-swe/shared/open-swe/manager/types";
+import { PlannerGraphState } from "@open-swe/shared/open-swe/planner/types";
+import { GraphState } from "@open-swe/shared/open-swe/types";
+
+export interface SessionCacheData {
+  plannerData?: { thread: Thread<PlannerGraphState>; run: null };
+  programmerData?: { thread: Thread<GraphState>; run: null };
```

Comment:
if run is always null just drop it?
```suggestion
  plannerData?: { thread: Thread<PlannerGraphState> };
  programmerData?: { thread: Thread<GraphState> };
```